docs/reference/modules/network.asciidoc

```diff
@@ -130,8 +130,25 @@ Enable or disable the https://en.wikipedia.org/wiki/Nagle%27s_algorithm[TCP no d
 setting. Defaults to `true`.
 
 `network.tcp.keep_alive`::
-Enable or disable https://en.wikipedia.org/wiki/Keepalive[TCP keep alive].
-Defaults to `true`.
+Configures the `SO_KEEPALIVE` option for this socket, which
+determines whether it sends TCP keepalive probes.
+
+`network.tcp.keep_idle`:: Configures the `TCP_KEEPIDLE` option for this socket, which
+determines the time in seconds that a connection must be idle before
+starting to send TCP keepalive probes. Defaults to `-1` which means to use
+the system default. May not be greater than 300. Only applicable on Linux and macOS,
+and requires Java 11 or newer.
+
+`network.tcp.keep_interval`:: Configures the `TCP_KEEPINTVL` option for this socket,
+which determines the time in seconds between sending TCP keepalive probes.
+Defaults to `-1` which means to use the system default. May not be greater than 300.
```

Comment:
```suggestion
Defaults to `-1` which means to use the smaller of 300 or the system default. May not be greater than 300.
```

Reply:
Ok, no, I understood #59278 ok and misunderstood this change. This suggestion isn't right.